plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeConnectorTest.java

```diff
@@ -1753,4 +1754,257 @@ private Set<String> getAllFilesFromCdfDirectory(String tableName)
                 .filter(path -> path.contains("/" + CHANGE_DATA_FOLDER_NAME))
                 .collect(toImmutableSet());
     }
+
+    @Test
+    public void testPartitionFilterQueryNotDemanded()
+    {
+        try (TestTable table = new TestTable(
+                getQueryRunner()::execute,
+                "test_partition_filter_not_demanded",
+                "(x VARCHAR, part VARCHAR) WITH (PARTITIONED_BY = ARRAY['part'])",
+                ImmutableList.of("'a', 'part_a'", "'b', 'part_b'"))) {
+            assertQuery("SELECT * FROM %s WHERE x='a'".formatted(table.getName()), "VALUES('a', 'part_a')");
+            assertQuery("SELECT * FROM %s WHERE part='part_a'".formatted(table.getName()), "VALUES('a', 'part_a')");
+        }
+    }
+
+    @Test
+    public void testQueryWithoutPartitionOnNonPartitionedTableNotDemanded()
+    {
+        String tableName = "test_no_partition_table_" + randomNameSuffix();
+        Session session = sessionWithPartitionFilterRequirement();
+
+        assertUpdate(session, "CREATE TABLE %s (x VARCHAR, part VARCHAR)".formatted(tableName));
+
+        assertUpdate(session, "INSERT INTO " + tableName + "(x, part) VALUES ('a', 'part_a'), ('b', 'part_b')", 2);
```

Comment:
Can we use test table ? Instead of `getQueryRunner()::execute` - query -> getQueryRunner().execute(session, sql)`

Reply:
And for other test cases ?